eppo/src/main/java/cloud/eppo/android/ConfigurationStore.java

```diff
@@ -49,19 +52,20 @@ public void loadFromCache(CacheLoadCallback callback) {
         AsyncTask.execute(() -> {
             Log.d(TAG, "Loading from cache");
             try {
-                RandomizationConfigResponse configResponse;
-                synchronized (cacheFile) {
-                    BufferedReader reader = cacheFile.getReader();
-                    configResponse = gson.fromJson(reader, RandomizationConfigResponse.class);
-                    reader.close();
-                }
+                RandomizationConfigResponse configResponse = readCacheFile();
                 if (configResponse == null || configResponse.getFlags() == null) {
-                    throw new JsonSyntaxException("Configuration file missing flags");
+                    throw new JsonSyntaxException("Cached configuration file missing flags");
                 }
-                flags = configResponse.getFlags();
-                if (flags.isEmpty()) {
+                if (configResponse.getFlags().isEmpty()) {
                     throw new IllegalStateException("Cached configuration file has empty flags");
                 }
+                if (loadedFromFetchResponse.get()) {
+                    Log.w(TAG, "Configuration already updated via fetch; ignoring cache");
+                    callback.onCacheLoadFail();
```

Comment:
Considering this a cache failure. Conceptually it sort of is as the cache was too slow 😛